src/gui.rs

```diff
@@ -394,195 +375,26 @@ impl Sam {
         }
 
         // Top menu bar
-        egui::TopBottomPanel::top("menubar").min_height(30.0).max_height(30.0).show(ctx, |ui| {
-            ui.set_enabled(!self.archive_window_open);
-            ui.horizontal_centered(|ui| {
-                let image = if ui.style().visuals.dark_mode {
-                    Image::new(egui::include_image!("../assets/logo_dark_mode.png"))
-                } else {
-                    Image::new(egui::include_image!("../assets/logo_light_mode.png"))
-                };
-                ui.add(image.max_size(Vec2::new(ui.available_width(), 20.0)));
-
-                ui.separator();
-                egui::widgets::global_dark_light_mode_switch(ui);
-                ui.separator();
-
-                ui.selectable_value(&mut self.tab, GuiTab::Launch, "🚀 Launch (F1)");
-                ui.selectable_value(&mut self.tab, GuiTab::Plot, "📈 Plot (F2)");
-                ui.selectable_value(&mut self.tab, GuiTab::Configure, "⚙  Configure (F3)");
-
-                ui.separator();
-
-                // Opening files manually is not available on web assembly
-                #[cfg(target_arch = "x86_64")]
-                if ui.selectable_label(false, "🗁  Open Log File").clicked() {
-                    if let Some(data_source) = open_log_file() {
-                        self.open_log_file(data_source);
-                    }
-                }
-
-                // Toggle archive panel
-                ui.toggle_value(&mut self.archive_window_open, "🗄 Flight Archive");
-
-                // Toggle archive panel
-                if ui.selectable_label(self.data_source.simulation_settings().is_some(), "💻 Simulate").clicked() {
-                    self.data_source = Box::new(SimulationDataSource::default());
-                }
-
-                // Show a button to the right to close the current log/simulation and go back to
-                // live view
-                ui.allocate_ui_with_layout(ui.available_size(), Layout::right_to_left(Align::Center), |ui| {
-                    if self.data_source.is_log_file() || self.data_source.simulation_settings().is_some() {
-                        if ui.button("❌").clicked() {
-                            self.close_data_source();
-                        }
-                    }
-                });
-            });
-        });
+        create_top_menu_bar(self, ctx);
```

Comment:
I have to say I'm not the biggest fan of passing `self` around to other files because that might make it more difficult in the future to tell where the contents of `Sam` are being modified. On the other hand I can see why it would be easier this way.

I think we could pass in something for `set_enabled` (and this would probably be done similarly for the other panels), and a reference to the data source, but for everything where we change state from within the panel, we might have to implement a response system similar to other egui widgets. So for instance having the function/method return some sort of enum like:

```rust
enum TopBarResponse {
    ArchiveWindowOpened,
    TabSelected(GuiTab),
    // etc
}
```

That's a bit more involved, so I think it may not be necessary to do that right now, but I think long-term this would be the cleanest solution, even though It would mean a bit more code in `src/gui.rs`.